Unity/BananaClient/TcpAdapter.cs

```diff
@@ -155,25 +163,43 @@ private int writePrimitiveBytes(byte[] dest, int offset, ulong data)
         }
 
         private void Receiving() {
-            try {
+            try 
+            {
                 // Waiting for sendingThread to establish the connnection.
                 while (running && error == "" && networkStream == null)
                     Thread.Sleep(0);
 
+                ushort eventType = 0;
                 while (running && error == "") {
-                    ushort eventType = ReadUInt16(networkStream);
-                   
-                    if (eventType == NetworkMessage.EventType_ConnectServer) {
+                    try
+                    {
+                        eventType = ReadUInt16(networkStream);
+                    }
+                    catch (Exception)
+                    {
+                        Thread.Sleep(100);
+                        continue;
+                    }
```

Comment:
Why is "try catch" needed? My understanding is that "ReadUInt16" will block until there is an UInt16 as long as the connection is open.

Reply:
-1 in memorystream means end of stream (relative to the position). This is different from network, we can update the UMT wrapper to handle it.

Reply:
In what case does "end of stream" happen? It is not clear to me according to linux doc: https://man7.org/linux/man-pages/man2/read.2.html. Does it make sense to keep retry for Error code like WSAECONNRESET? Shouldn't we terminate the adapter and notifying the application layer in that case?

Reply:
Here is the doc for c# which may be more propriate: https://docs.microsoft.com/en-us/windows/win32/winsock/windows-sockets-error-codes-2

Reply:
Yeah you are right, the try / catch is really for memorystream in which -1 is a simple readptr reaching end of stream. In a network stream the read simply hangs unless the connection is closed. It is more appropriate to shutdown / or attempt to enter a reconnecting state upon a -1.